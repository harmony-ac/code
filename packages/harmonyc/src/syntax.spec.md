# Syntax

Markdown code is mapped to the Harmony test model along the following rules.

## Empty file

- empty => empty test design
- there are empty lines => empty test design

  ```markdown

  ```

- there is only a paragraph => empty test design

  ```markdown
  The user should be able to log in.
  ```

## Section headings

- there is only a h1 => one section

  ```markdown
  # Login form
  ```

- there are more h1s => forked sections

  ```markdown
  # Login

  # Logout
  ```

- there are only h2s => forked sections

  ```markdown
  ## Login

  ## Logout
  ```

- there are nested headings => nested forked sections

  ```markdown
  # Login form

  ## Log in
  ```

- there are deep nested headings => deep nested forked sections

  ```markdown
  # Login form

  ## Log in

  ### Empty checks

  ### Email validation

  ## Log out

  ### Re-login
  ```

## Bulleted lists

- there is only a bullet point => one step

  ```markdown
  - log in
  ```

- there are more bullet points => forked steps
  ```markdown
  - log in
  - log out
  ```
- there are nested bullet points => nested forked steps
  ```markdown
  - log in
    - create order
    - log out
  - sign up
    - log out
  ```
- deep nested bullet points => deep nested forked steps

## Numbered lists

- there is only a numbered item => one step, not forked
  ```markdown
  1. log in
  ```

- there are more numbered items => sequential steps
  ```markdown
  1. log in
  2. log out
  ```

## Responses

- there is only response => step with one response
  ```markdown
  - => logged out
  ```
- there is action and response => step with action and response
  ```markdown
  - log out => logged out
  ```
- there are more responses => step with more responses
  ```markdown
  - log out => logged out => on home page
  ```

## Headings within lists

- there is a heading in list item => one section
  ```markdown
  - # Login form
  ```
- there are headings in bulleted items => forked sections
  ```markdown
  - ### Login
  - ### Logout
  ```

## Docstrings

- there is a docstring => action with docstring
  ````markdown
  - action
    ```
    Docstring
    ```
  ````
