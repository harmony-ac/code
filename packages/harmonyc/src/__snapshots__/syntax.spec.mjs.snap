// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`T1 - Syntax - Empty file > empty test design 1`] = `[]`;

exports[`T2 - Syntax - Empty file > empty test design 1`] = `[]`;

exports[`T3 - Syntax - Empty file > empty test design 1`] = `[]`;

exports[`T4 - Syntax - Section headings > one section 1`] = `[]`;

exports[`T5 - Syntax - Section headings > forked sections 1`] = `[]`;

exports[`T6 - Syntax - Section headings > nested forked sections 1`] = `[]`;

exports[`T7 - Syntax - Section headings > deep nested forked sections 1`] = `[]`;

exports[`T8 - Syntax - Bulleted lists > one step 1`] = `[]`;

exports[`T9 - Syntax - Bulleted lists > forked steps 1`] = `[]`;

exports[`T10 - Syntax - Bulleted lists > nested forked steps 1`] = `[]`;

exports[`T11 - Syntax - Bulleted lists > deep nested forked steps 1`] = `[]`;

exports[`T12 - Syntax - Numbered lists > one step, not forked 1`] = `[]`;

exports[`T13 - Syntax - Numbered lists > sequential steps 1`] = `[]`;

exports[`T14 - Syntax - Responses > step with one response and empty action 1`] = `[]`;

exports[`T15 - Syntax - Responses > step with more responses 1`] = `[]`;

exports[`T16 - Syntax - Headings within lists > one section 1`] = `[]`;

exports[`T17 - Syntax - Docstrings > action with docstring 1`] = `[]`;

exports[`T19 - Syntax - Section headings > forked sections 1`] = `[]`;

exports[`T20 - Syntax - Numbered lists > nested sequences and forks 1`] = `[]`;

exports[`T21 - Syntax - Numbered lists > bulleted items are forks, numbered items not 1`] = `[]`;

exports[`T22 - Syntax - Responses > step with action and response 1`] = `[]`;

exports[`T23 - Syntax - Headings within lists > forked sections 1`] = `[]`;
